zerver/actions/message_edit.py

```diff
@@ -820,6 +820,21 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
 
     send_event(user_profile.realm, event, users_to_be_notified)
 
+    if (
+        topic_name is not None
+        and new_stream is None
+        and content is None
+        and len(changed_messages) > 0
+    ):
+        assert stream_being_edited is not None
+        maybe_send_resolve_topic_notifications(
+            user_profile=user_profile,
+            stream=stream_being_edited,
+            old_topic=orig_topic_name,
+            new_topic=topic_name,
+            changed_messages=changed_messages,
+        )
```

Comment:
Two comments on the commit message for this first commit:
* The `"Fixes #21712 partially"` commit message would close the issue as fixed as soon as this is merged, because GitHub's regex is sloppy and just looks at whether the string "Fixes #[0-9]+" appears in the commit message. 
"Fixes part of ..." is a workaround, though I did a more careful wording.
* We should explain that this commit does change behavior; as described in my revised commit message, the behavior change is likely equally good / a bit better.